WordVectors/python/analysis_notebook.ipynb

```diff
@@ -0,0 +1,892 @@
+   "source": [
+    "# create a scatter plot of the projection\n",
+    "\n",
+    "# set the points for the x_axis by grabbing all the items in column one of the results variable\n",
+    "x_axis = result[:,0]\n",
+    "\n",
+    "# set the points for the y_axis by grabbing all the items in column two of the results variable\n",
+    "y_axis = result[:,1]\n",
+    "\n",
+    "# create a new figure of size (8, 6)\n",
+    "fig = plt.figure(figsize=(8, 6)) # you can change this size\n",
+    "\n",
+    "# create a variable for the axes and let the computer know we want a 2D plot\n",
+    "ax = fig.add_subplot(111)\n",
+    "\n",
+    "# scatterplot the points by using the x_axis variable as X and the y_axis variable as Y\n",
+    "ax.scatter(x_axis, y_axis)\n",
+    "\n",
+    "# iterate through our labels list\n",
+    "    for i in range(len(labels)):\n",
+    "        \n",
+    "        # assign labels to the points and make those labels italic, blue, and size 10\n",
+    "        ax.text(x_axis[i], y_axis[i], labels[i], style ='italic',\n",
+    "        fontsize = 10, color =\"blue\")  # you can change these values\n",
+    "        \n",
+    "# label the x axis\n",
+    "ax.set_xlabel('x axis') # you can change this\n",
+    "\n",
+    "# label the y axis\n",
+    "ax.set_ylabel('y axis') # you can change this\n",
+    "   \n",
+    "# visualize the plot\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The code for the `pca()` function in its entirety with the correct indention is below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "labels = list(model.wv.key_to_index)\n",
+    "def pca(model):\n",
+    "    X = model.wv[model.wv.key_to_index]  # get all the vectors\n",
+    "    pca = PCA(n_components=2)\n",
+    "    result = pca.fit_transform(X)\n",
+    "    # create a scatter plot of the projection\n",
+    "    x_axis = result[:,0]\n",
+    "    y_axis = result[:,1]\n",
+    "    fig = plt.figure(figsize=(8, 6))\n",
+    "    ax = fig.add_subplot(111)\n",
+    "    ax.scatter(x_axis, y_axis)\n",
+    "    for i in range(len(labels)):\n",
+    "        ax.text(x_axis[i], y_axis[i], labels[i], style ='italic',\n",
+    "        fontsize = 10, color =\"blue\")\n",
+    "    ax.set_xlabel('x axis')\n",
+    "    ax.set_ylabel('y axis')\n",
+    "\n",
+    "    \n",
+    "    plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As you can see, labeling every point on the PCA plot can make actually reading the labels quite difficult. Instead of labeling every point, we might want to plot all of our points and visualize a random number of them. In order to accomplish this, we are going to use the python library `random`. We are going to insert the code below into our `pca()` function to replace the `for` loop we previously used to label. \n",
```

Comment:
Running the code block above only defines the function, it doesn’t run it. “As you can see” really doesn’t apply with the code block as-is! Could you add a call to `pca()` to demonstrate?